src/sagas/index.js

```diff
@@ -18,6 +49,50 @@ function* watchImport() {
   }
 }
 
+function* watchSaveRange() {
+  while (true) {
+    const { payload } = yield take(UPDATE_RANGE);
+    const { startTime, endTime } = payload;
+    if (startTime) {
+      yield put(movePoint(startTime.x, startTime.originalX));
+    }
+    if (endTime) {
+      yield put(movePoint(endTime.x, endTime.originalX));
+    }
+    yield put(editMetadata(null));
+  }
+}
+
+function* watchResetDocument() {
+  while (true) {
+    yield take(RESET_DOCUMENT);
+    const duration = yield select(getDuration);
+    yield put(loadRanges(duration));
+  }
+}
+
+function* watchPreviousBubble() {
+  while (true) {
+    yield take(NEXT_BUBBLE);
+    const nextBubbleTime = yield select(getNextBubbleStartTime);
+    yield put(setCurrentTime(nextBubbleTime));
+  }
+}
+
+function* watchNextBubble() {
+  while (true) {
+    yield take(PREVIOUS_BUBBLE);
+    const previousBubbleTime = yield select(getPreviousBubbleStartTime);
+    yield put(setCurrentTime(previousBubbleTime));
+  }
+}
+
 export default function* root() {
-  yield all([fork(watchImport)]);
+  yield all([
+    fork(watchImport),
```

Comment:
you can also use `takeEvery(ACTION_NAME, watchImport)` and that will send the the action as the first argument of the generator (and means you can drop the `while(true) { ... }`).

Reply:
https://github.com/digirati-co-uk/timeliner/pull/105/commits/7435eeeefcac13ead3db362540bf7061c42ef40d